### PR TITLE
WSO2 G-reg smart bear plugin

### PR DESCRIPTION
**Pull request by manojgunayadev**

### CODE REVIEW

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+                selectionResult, project);
+        try {
+            worker.waitDialog.run(worker);
+        } catch (Exception e) {
+            UISupport.showErrorMessage(e.getMessage());
+            log.error(e);
+        }
+        if (worker.addedRestServices != null && worker.addedRestServices.size() > 0) {
+            return worker.addedRestServices;
+        } else if (worker.addedSoapServices !=null && worker.addedSoapServices.size()> 0){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+            }
+            RestService[] restServices;
+            WsdlInterface[] wsdlInterfaces;
+            try {
+
+                boolean isSwagger2 = false;
+
+                // We are importing the Resource definition from the swagger resource here.
+                // Once imported, we get an array of RestServices as the result.
+                if (apiInfo.getResType().equals(ResourceConstants.RESOURCE_TYPE_SWAGGER)){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue. There should be a space in between ) and {

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+                    if (restServices != null) {
+                        if (isTestSuiteRequired) {
+                            createTestSuiteForResetServices(restServices, isSwagger2,apiInfo);
+                        }
+                    }
+                    // We set the Auth Profile here.
+                    setAuthProfile();
+                }else{
+                    wsdlInterfaces = Utils.importSOAPServiceToProject(apiInfo, project);
+                    if (addedSoapServices !=null &&  isTestSuiteRequired){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue. There should be a space in between ) and {

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+                errors = String.format("Failed to read Resource description for[%s] - [%s]",
+                        apiInfo.getName(), e.getMessage());
+                continue;
+            }
+            if (restServices != null) {
+                addedRestServices.addAll(Arrays.asList(restServices));
+            }
+
+            if (wsdlInterfaces !=null){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue. There should be a space in between ) and {

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            errors += "\nPlease contact WSO2 support for assistance";
+        }
+
+        return null;
+    }
+
+    private void deleteSwaggerContentFile(String filePath) {
+        try{
+            Files.delete(Paths.get(filePath));
+        }catch (IOException ioEx){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issues.
1. There should be a space in between ) and catch block
2. There should be a space in between ) and { characters

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+            log.error("Error occured deleting swagger content temp file " +filePath + ioEx);
+        }
+    }
+
+    private void createTestSuiteForResetServices(RestService[] restServices, boolean isSwagger2,
+                                                 ResourceInfo apiInfo){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue. There should be a space in between ) and {

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+                if (isLoadTestSelected) {
+                    if (testCase != null) {
+                        testCase.addNewLoadTest(testCase.getName());
+                    }
+                }
+            }
+        }
+    }
+
+    private void createTestSuiteForSoapServices(WsdlInterface[] wsdlInterfaces){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue. There should be a space in between ) and {

File: src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+                ResourceConstants.LOGIN_PROFILE_FILE_PREFIX + ResourceConstants.LOGIN_PROFILE_FILE_SUFFIX);
+
+        Map profileMap = new HashMap();
+
+        if (loginPrfileSettingsFile.exists()) {
+
+            try (
+                 FileInputStream fis = new FileInputStream(loginPrfileSettingsFile);
+                 BufferedReader br = new BufferedReader(new InputStreamReader(fis));
+            ){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java]: Formatting issue. There should be a space in between ) and {

File: src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+                String line;
+
+                while ((line = br.readLine()) != null) {
+                    try {
+                        profileMap.put(line.split(":")[0], line.split(":")[1]);
+                    }catch(PatternSyntaxException patEx){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java]: Formatting issues. 
1. There should be a space in between ) and catch block
2. There should be a space in between ) and { characters

File: src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+                        log.error("Pattern syntax error in setting profile data"+line,patEx);
+                    }catch (ArrayIndexOutOfBoundsException arrEx){
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java]: Formatting issues.
1. There should be a space in between ) and catch block
2. There should be a space in between ) and { characters

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+                    isSwagger2 = checkIsSwagger2(apiInfo, isSwagger2);
+
+                    if (restServices != null) {
+                        if (isTestSuiteRequired) {
+                            createTestSuiteForResetServices(restServices, isSwagger2,apiInfo);
+                        }
+                    }
+                    // We set the Auth Profile here.
+                    setAuthProfile();
+                }else{
```
Comment (cnapagoda) [inline src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Formatting issue

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 1 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 3 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 4 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 5 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 6 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 7 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

File: src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java (hunk 8 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java]: Done

File: src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java (hunk 9 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java]: Done

File: src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java (hunk 10 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/workspace/WSO2GregWorkspace.java]: Done

File: src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy (hunk 11 above)
Comment (manojgunayadev) [reply src/main/java/org/wso2/greg/plugin/worker/ResourceImporterWorker.groovy]: Done

**Pull request merged**